File: 1637.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    int n;
    cin >> n;

    vector<vector<int>> points(n, vector<int>(2));

    for (int i = 0; i < n; i++) {
        cin >> points[i][0] >> points[i][1];
    }

    vector<int> x;

    for (int i = 0; i < n; i++) {
        x.push_back(points[i][0]);
    }

    sort(x.begin(), x.end());

    int ans = 0;

    for (int i = 1; i < x.size(); i++) {
        ans = max(ans, x[i] - x[i - 1]);
    }

    cout << ans;
}

File: 1637.py
n = int(input())

points = []
for _ in range(n):
    points.append(list(map(int, input().split())))

x = []

for point in points:
    x.append(point[0])

x.sort()

mx = 0
for i in range(1, len(x)):
    mx = max(mx, x[i] - x[i-1])

print(mx)